### PR TITLE
INTMDB-410: PrivateLink Endpoint Examples + Documentation Cleanup

### PR DESCRIPTION
**Pull request by evertsd**
## Description

To help prevent customers from running into issues where cluster/advanced_cluster is created before privatelink_endpoint, leaving their cluster/advanced cluster resources without the respective connection strings.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code

### CODE REVIEW

**Comment by evertsd** [comment]
Some of the files here are listed as deleted, they've simply been moved/refactored and that isn't being recognized:
- examples/aws-atlas-privatelink-regionalized/ => examples/aws-privatelink-endpoint/cluster-geosharded/
- examples/aws-atlas-privatelink/ => examples/aws-privatelink-endpoint/cluster

`examples/aws-privatelink-endpoint/` now contains 3 examples for replication cluster, geosharded cluster (regionalized privatelink endpoints), and serverless instance.

These three examples have been verified as a part of this work, with READMEs explaining what is necessary to get proper connection strings on initial `terraform apply`, as well as why through the lens of resource dependency chains.

**Comment by Zuhairahmed** [comment]
Thanks @evertsd can you double check automated GitHub test fails?

**Comment by evertsd** [comment]
Yep @Zuhairahmed looks like a small issue in one of the examples where a declared variable for `instance_name` was unused. I've fixed the example to use that `instance_name`

File: mongodbatlas/data_source_mongodbatlas_serverless_instance.go [hunk 1]
```diff
@@ -50,6 +50,12 @@ func dataSourceMongoDBAtlasServerlessInstanceRead(ctx context.Context, d *schema
 		return diag.Errorf("error setting `connection_strings_standard_srv` for serverless instance (%s): %s", d.Id(), err)
 	}
 
+	if len(serverlessInstance.ConnectionStrings.PrivateEndpoint) > 0 {
+		if err := d.Set("connection_strings_private_endpoint_srv", flattenSRVConnectionString(serverlessInstance.ConnectionStrings.PrivateEndpoint)); err != nil {
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_serverless_instance.go:54]: are these changes wanted? I thought this PR was updating only documentation

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by Zuhairahmed — APPROVED** [review]
LGTM, great job referencing the examples in each of the docs as well @evertsd! Would just address @andreaangiolillo question above and after that feel free to merge.

**Comment by evertsd** [comment]
Ah, yes @Zuhairahmed and @andreaangiolillo I forgot about that.

This PR ended up not just being documentation. 
While writing my example for `serverless-instance`, I found that the only way to create the resources and retrieve the connection string in one `terraform apply`, would be to source that connection string from a data source.

The one wrinkle with that plan was that the data source didn't currently support the connection string. I'll ping @martinstibbe  as well to verify that there wasn't a reason that attribute was left off of the data source in the first place.

**Comment by Zuhairahmed** [comment]
> The one wrinkle with that plan was that the data source didn't currently support the connection string. I'll ping @martinstibbe as well to verify that there wasn't a reason that attribute was left off of the data source in the first place.

Got it, if this behavior is correct would just make sure to call in docs for both resource and data source for `serverless-instance`

**Comment by evertsd** [comment]
Alright @Zuhairahmed, I think that latest commit provides a little more clarity in the serverless-instance documentation around this use case

**Comment by Zuhairahmed** [comment]
> Alright @Zuhairahmed, I think that latest commit provides a little more clarity in the serverless-instance documentation around this use case

LGTM, @martinstibbe can you review this one too?

**Review by martinstibbe — APPROVED** [review]
LGTM

**Pull request merged**